### PR TITLE
feat: Adding `files` support for file sending in context

### PR DESCRIPTION
**Pull request by B1ue-Dev**
## About

Title.

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

**Review by sillyElvy — CHANGES_REQUESTED** [review]
Docs should be updated

**Comment by B1ue-Dev** [comment]
> Docs should be updated

Oh it.

File: interactions/api/http/interaction.py [hunk 1]
```diff
@@ -248,21 +276,44 @@ async def get_original_interaction_response(
...
+            part = file_data.append_json(data)
+            part.set_content_disposition("form-data", name="payload_json")
+            data = None
+
+            for id, file in enumerate(files):
+                part = file_data.append(
+                    file._fp,
+                )
+                part.set_content_disposition(
+                    "form-data", name=f"files[{str(id)}]", filename=file._filename
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
                    "form-data", name=f"files[{id}]", filename=file._filename
```

File: interactions/api/http/interaction.py [hunk 2]
```diff
@@ -283,15 +334,39 @@ async def delete_interaction_response(
...
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
                    "form-data", name=f"files[{id}]", filename=file._filename
```

File: interactions/api/http/interaction.py [hunk 3]
```diff
@@ -248,21 +276,44 @@ async def get_original_interaction_response(
         )
 
     async def edit_interaction_response(
-        self, data: dict, token: str, application_id: str, message_id: str = "@original"
+        self,
+        data: dict,
+        files: Optional[List[File]],
+        token: str,
+        application_id: str,
+        message_id: str = "@original",
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        data: dict,
        token: str,
        application_id: str,
        files: Optional[List[File]] = None,
        message_id: str = "@original",
```

File: interactions/api/http/interaction.py [hunk 4]
```diff
@@ -1,6 +1,10 @@
 from typing import TYPE_CHECKING, List, Optional, Union
 
+from aiohttp import MultipartWriter
+
+from ...utils.missing import MISSING
```
Comment (Damego) [inline interactions/api/http/interaction.py]: We don't use `MISSING` in the http files

File: interactions/api/http/interaction.py [hunk 5]
```diff
@@ -216,17 +220,41 @@ async def get_all_application_command_permissions(
         )
 
     async def create_interaction_response(
-        self, token: str, application_id: int, data: dict
+        self,
+        token: str,
+        application_id: int,
+        data: dict,
+        files: Optional[List[File]] = MISSING,
```
Comment (Damego) [inline interactions/api/http/interaction.py]: We don't use `MISSING` in the http files

File: interactions/api/http/interaction.py [hunk 6]
```diff
@@ -248,21 +276,44 @@ async def get_original_interaction_response(
...
         :param data: A dictionary containing the new response.
+        :param files: The files to send.
         :param token: the token of the interaction
         :param application_id: Application ID snowflake.
         :param message_id: Message ID snowflake. Defaults to `@original` which represents the initial response msg.
         :return: Updated message data.
         """
         # ^ again, I don't know if python will let me
+        file_data = None
+        if files is not MISSING and files is not None:
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        if files is not None:
```

File: interactions/api/http/interaction.py [hunk 7]
```diff
@@ -248,21 +276,44 @@ async def get_original_interaction_response(
...
+        message_id: str = "@original",
     ) -> dict:
         """
         Edits an existing interaction message, but token needs to be manually called.
 
         :param data: A dictionary containing the new response.
+        :param files: The files to send.
         :param token: the token of the interaction
         :param application_id: Application ID snowflake.
         :param message_id: Message ID snowflake. Defaults to `@original` which represents the initial response msg.
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        :param data: A dictionary containing the new response.
        :param token: the token of the interaction
        :param application_id: Application ID snowflake.
        :param files: The files to send.
        :param message_id: Message ID snowflake. Defaults to `@original` which represents the initial response msg.
```

File: interactions/api/http/interaction.py [hunk 8]
```diff
@@ -283,15 +334,39 @@ async def delete_interaction_response(
             Route("DELETE", f"/webhooks/{int(application_id)}/{token}/messages/{message_id}")
         )
 
-    async def _post_followup(self, data: dict, token: str, application_id: str) -> dict:
+    async def _post_followup(
+        self,
+        data: dict,
+        files: Optional[List[File]],
+        token: str,
+        application_id: str,
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        data: dict,
        token: str,
        application_id: str,
        files: Optional[List[File]] = None,
```

File: interactions/api/http/interaction.py [hunk 9]
```diff
@@ -283,15 +334,39 @@ async def delete_interaction_response(
...
         Send a followup to an interaction.
 
         :param data: the payload to send
+        :param files: the files to send
         :param application_id: the id of the application
         :param token: the token of the interaction
         """
 
+        file_data = None
+        if files is not MISSING and files is not None:
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        if files is not None:
```

File: interactions/api/http/interaction.py [hunk 10]
```diff
@@ -283,15 +334,39 @@ async def delete_interaction_response(
...
+        token: str,
+        application_id: str,
+    ) -> dict:
         """
         Send a followup to an interaction.
 
         :param data: the payload to send
+        :param files: the files to send
         :param application_id: the id of the application
         :param token: the token of the interaction
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        :param data: the payload to send
        :param application_id: the id of the application
        :param token: the token of the interaction
        :param files: the files to send
```

**Review by Damego — CHANGES_REQUESTED** [review]

File: interactions/api/http/interaction.py (hunk 5 above)
Comment (B1ue-Dev) [reply interactions/api/http/interaction.py]: Applied.

File: interactions/api/http/interaction.py (hunk 4 above)
Comment (B1ue-Dev) [reply interactions/api/http/interaction.py]: Applied.

File: interactions/api/http/interaction.py [hunk 11]
```diff
@@ -248,21 +275,44 @@ async def get_original_interaction_response(
...
         :param data: A dictionary containing the new response.
         :param token: the token of the interaction
         :param application_id: Application ID snowflake.
+        :param files: The files to send.
         :param message_id: Message ID snowflake. Defaults to `@original` which represents the initial response msg.
         :return: Updated message data.
         """
         # ^ again, I don't know if python will let me
+        file_data = None
+        if files is not None:
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        if files:
```

File: interactions/api/http/interaction.py [hunk 12]
```diff
@@ -283,15 +333,39 @@ async def delete_interaction_response(
...
         Send a followup to an interaction.
 
         :param data: the payload to send
         :param application_id: the id of the application
         :param token: the token of the interaction
+        :param files: the files to send
         """
 
+        file_data = None
+        if files is not None:
```
Comment (Damego) [inline interactions/api/http/interaction.py]:
```suggestion
        if files:
```

**Review by Damego — APPROVED** [review]
Looks good to me!

**Review by sillyElvy — APPROVED** [review]

**Pull request merged**